### DISCUSSION TITLE
Narrowing search results to specific types

### DISCUSSION THREAD

**Discussion by adrian-hebda** [osm-search/Nominatim#3161] [Q&A]
Hello,

I would like to narrow down the search results using `q=<query>` in such a way that the results returned are only cities (city+village+town). I am using the option to narrow the results to a country area using `countrycodes=<countrycode>[,<countrycode>][,<countrycode>]` and in the country area, I would like to find only cities. Is there such an option? 

I saw that it used to be possible to get a similar result using the `featuretype `parameter, but this has been removed(?). Is there any alternative to it?

**Comment by lonvia** [comment]
You can use `featureType=city` (note the upper-case T). The parameter was always poorly documented and might have yielded surprising results until previously. With the new frontend implementation that is now live on nominatim.openstreetmap.org, it should do exactly what you want (subject to Nominatim properly recognising a city/town/village which can be surprisingly hard sometimes with OSM data).

Updating the online documentation is very high on my list.

**Reply by adrian-hebda** [reply]
Are there some other values besides city for the featureType parameter?

**Reply by lonvia** [reply]
See https://github.com/osm-search/Nominatim/blob/517a0cb67397949f3c377e8d7f93bcb239929967/nominatim/api/v1/helpers.py#L40-L45

The numbers refer to [address ranks](https://nominatim.org/release-docs/develop/customize/Ranking/#address-rank).

**Comment by estherlusky** [comment]
is it possible to get results only for cities not for villages and towns?
